data/provider/src/main/java/tech/pegasys/artemis/provider/JsonProvider.java

```diff
@@ -114,7 +113,16 @@ public boolean shouldSkipClass(Class<?> clazz) {
               return Bitvector.fromBytes(bytes, length);
             });
 
-    Gson gson = builder.create();
-    return gson.toJson(state);
+    builder.registerTypeAdapter(
+        BLSSignature.class,
```

Comment:
Couldn't we register another type adapter here for `PeerID` and then not need the various `getPeerString` type methods at all?